test/unit/dom-if.html

```diff
@@ -497,19 +504,132 @@
 
     });
 
-    suite('queueing race conditions', function() {
+    suite('attach/detach tests', function() {
+
+      test('remove, append domif', function(done) {
+        var domif = document.querySelector('#simple');
+        domif.if = true;
+        outerContainer.removeChild(domif);
+        setTimeout(function() {
+          var clients = outerContainer.querySelectorAll('x-client');
+          assert.equal(clients.length, 0);
+          outerContainer.appendChild(domif);
+          setTimeout(function() {
+            var clients = outerContainer.querySelectorAll('x-client');
+            assert.equal(clients[0].uid, 0);
+            assert.equal(clients[1].uid, 1);
+            assert.equal(clients[2].uid, 2);
+            assert.equal(clients[1].previousElementSibling, clients[0]);
+            assert.equal(clients[2].previousElementSibling, clients[1]);
+            assert.equal(domif.previousElementSibling, clients[2]);
```

Comment:
We tend not to favor making indirections like this specifically in tests. The idea is to  try to make the asserts as straightforward as possible.